### PR TITLE
PP-11224: Add Dockerfile, config, and unit test workflow

### PR DESCRIPTION
**Pull request by jfharden**
Add initial Dockerfile, config, and a workflows to test and tag releases.

You can test this locally by going into the tests/ directory and running `./test.sh` (You will need jq installed)

### CODE REVIEW

**Comment by jfharden** [comment]
I tried to add otel-config-validator to test the config however....

1. it doesn't understand the sigv4auth extension so sees it as an error (opened https://github.com/lightstep/otel-config-validator/issues/8 to suggest it as a feature)
2. It doesn't exit with a bad exitcode when validation fails so it's a bit fragile to detect the errors (opened https://github.com/lightstep/otel-config-validator/issues/9 with a feature request to add a fail on error flag to the cli)

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  build-and-test:
+    name: Container Build Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+      - name: Docker build
+        run: |
+          docker build -t pay-adot:github-tests .
```
Comment (katstevens) [inline .github/workflows/run-tests.yml:27]: Should we tag the build with the SHA ([like we do on the telegraf tests](https://github.com/alphagov/pay-telegraf/blob/main/.github/workflows/run-tests.yml#L28))?

Comment (jfharden) [reply .github/workflows/run-tests.yml:27]: I don't think it's needed, we're not actually putting this image anywhere so the tag is completely ephemeral anyway

File: tests/node_service/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "express": "^4.18.2",
+    "prom-client": "^14.2.0"
+  }
+}
```
Comment (katstevens) [inline tests/node_service/package.json:15]: Can we add config for these npm packages into `dependabot.yml`? (The usual `open-pull-requests-limit: 0`)

Comment (jfharden) [reply tests/node_service/package.json:15]: Oh yeah, totally forgot about the npm deps (I wasn't going to have any until the otel config validator didn't work :()

Comment (jfharden) [reply tests/node_service/package.json:15]: Added in https://github.com/alphagov/pay-adot/pull/2/commits/094eabd0a24a92de11284461bf7353b82c4a13ab

File: tests/node_service/index.mjs [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import { randomUUID } from 'crypto'
```
Comment (katstevens) [inline tests/node_service/index.mjs]: Is this used?

**Review by katstevens — DISMISSED** [review]
Been playing around with the tests locally, looks good! I ran into a minor bit of confusion when trying to make the tests fail on purpose, but realised it was because docker-compose hadn't rebuilt the `test-app` image for me (and I had to delete it manually). In practice I think this will be fine - GHA will not have any cached images lying around.

File: tests/node_service/index.mjs (hunk 3 above)
Comment (jfharden) [reply tests/node_service/index.mjs]: Oh it's not anymore 🤔 I did have it previously and decided it was pointless. I'll remove this

**Comment by jfharden** [comment]
> Been playing around with the tests locally, looks good! I ran into a minor bit of confusion when trying to make the tests fail on purpose, but realised it was because docker-compose hadn't rebuilt the `test-app` image for me (and I had to delete it manually). In practice I think this will be fine - GHA will not have any cached images lying around.

Yeah I think I will add `docker-compose build` to the start of the test script. It will rebuild the images using the normal docker caching rules instead of not rebuilding at all

**Comment by jfharden** [comment]
@katstevens I've done the couple of updates (removed the unused import, added the docker-compose build). Would you mind re-reviewing?

**Review by katstevens — APPROVED** [review]

**Pull request merged**